tests/Common/BranchComponentTest.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace Keboola\Test\Common;
+
+use Keboola\StorageApi\ClientException;
+use Keboola\StorageApi\DevBranches;
+use Keboola\StorageApi\Options\Components\ListComponentConfigurationsOptions;
+use Keboola\StorageApi\Options\Components\ListComponentsOptions;
+use Keboola\Test\StorageApiTestCase;
+
+class BranchComponentTest extends StorageApiTestCase
+{
+    public function setUp()
+    {
+        parent::setUp();
+
+        $components = new \Keboola\StorageApi\Components($this->_client);
+        foreach ($components->listComponents() as $component) {
+            foreach ($component['configurations'] as $configuration) {
+                $components->deleteConfiguration($component['id'], $configuration['id']);
+            }
+        }
+
+        // erase all deleted configurations
+        foreach ($components->listComponents((new ListComponentsOptions())->setIsDeleted(true)) as $component) {
+            foreach ($component['configurations'] as $configuration) {
+                $components->deleteConfiguration($component['id'], $configuration['id']);
+            }
+        }
+    }
+
+    public function testGetComponentConfigurations()
+    {
+        $providedToken = $this->_client->verifyToken();
+        $devBranch = new DevBranches($this->_client);
+        // cleanup
+        $branchesList = $devBranch->listBranches();
+        $branchName = __CLASS__ . '\\' . $this->getName() . '\\' . $providedToken['id'];
+        $branchesCreatedByThisTestMethod = array_filter(
+            $branchesList,
+            function ($branch) use ($branchName) {
+                return strpos($branch['name'], $branchName) === 0;
+            }
+        );
+        foreach ($branchesCreatedByThisTestMethod as $branch) {
+            $devBranch->deleteBranch($branch['id']);
+        }
+
+        $branch = $devBranch->createBranch($branchName);
+
+        $branchComponents = new \Keboola\StorageApi\Components($this->getBranchAwareDefaultClient($branch['id']));
+        $componentId = 'transformation';
+        $configs = $branchComponents->listComponentConfigurations(
+            (new ListComponentConfigurationsOptions())->setComponentId($componentId)
+        );
+        $this->assertEmpty($configs);
```

Comment:
Budeme to přepisovat ještě a je to jen temp a je to jedno, ale obecně myslím, že by u toho měla být message proč je to očekávané.

Reply:
Bud t custom error mesage u asertu nebo comment